crates/gitbutler-tauri/src/main.rs

```diff
@@ -132,6 +131,25 @@ fn main() {
                     tauri_app.on_menu_event(move |_handle, event| {
                         menu::handle_event(&window.clone(), &event)
                     });
+
+                    if app_settings.get()?.clone().feature_flags.v3 {
+                        #[cfg(target_os = "macos")]
+                        use tauri::LogicalPosition;
+                        #[cfg(target_os = "macos")]
+                        use tauri_plugin_traffic_lights::WindowExt;
+                        #[cfg(target_os = "macos")]
+                        // NOTE: Make sure you only call this ONCE per window.
+                        {
+                            if let Some(window) = tauri_app.get_window("main") {
+                                #[cfg(target_os = "macos")]
+                                // NOTE: Make sure you only call this ONCE per window.
+                                window
+                                    .setup_traffic_lights_inset(LogicalPosition::new(16.0, 25.0))?;
+                            };
+                        }
```

Comment:
This is only called _once_ for the app atm, and doesn't reposition new lights in new windows. Also, it applies the same positioning to V2. Let's fix before we need to make a next release!

Reply:
@Byron looks like there was a reason for the 2nd call

Reply:
I think the issues @mtsgrd is describing would be true in multi-window mode and when V3 and V2 is switched while the app is running? 
Indeed it's made to only work if the app launches in V3 mode and it's assumed to only have one Window - multi-window support was never made official, and from what I remember it's currently hanging/deadlocking on Windows even. When V3 is switched to V2, the app needs a restart for the buttons to appear in the right spot.

Should there be a follow-up that makes this multi-window compatible?

> @Byron looks like there was a reason for the 2nd call

Maybe, but if it wasn't added by you or me, who would know about it?